### PR TITLE
EOPA - Batch Query API

### PR DESCRIPTION
**Pull request by philipaconrad**
## What changed?

This PR includes support for the Batch Query API in Enterprise OPA, and includes basic tests for the API. Enterprise OPA is spun up in a container as a test fixture, similar to how we've been running OPA in a container test fixture.

## TODOs

 - [x] Add `EOPA_LICENSE_KEY` for the CI workflows. (Used to spin up the EOPA test container.)
 - Test cases:
   - [x] All-Success
   - [x] No input case
   - [x] Mixed
   - [x] All-Failure
   
 ## Additional Resources

 - https://github.com/StyraInc/enterprise-opa/tree/main/openapi

### CODE REVIEW

File: test/SmokeTest.Tests/SmokeTest.Tests.csproj [hunk 1]
```diff
@@ -31,10 +31,7 @@
 
   <!-- Ensure our data files are copied next to the test executables. -->
   <ItemGroup>
-    <None Update="testdata\data.json" CopyToOutputDirectory="PreserveNewest" />
-    <None Update="testdata\policy.rego" CopyToOutputDirectory="PreserveNewest" />
-    <None Update="testdata\simple\system.rego" CopyToOutputDirectory="PreserveNewest" />
-    <None Update="testdata\weird_name.rego" CopyToOutputDirectory="PreserveNewest" />
+    <None Update="testdata\**\*" CopyToOutputDirectory="PreserveNewest" />
```
Comment (srenatus) [inline test/SmokeTest.Tests/SmokeTest.Tests.csproj:34]: Yay

**Comment by philipaconrad** [comment]
:information_source: Trying to get the test suite to run. :sweat_smile:

**Comment by philipaconrad** [comment]
:information_source: It looks like resolving the merge conflict in `.speakeasy/gen.lock` was sufficient to get the normal Pull Request jobs to kick off. :slightly_smiling_face:

File: test/SmokeTest.Tests/SmokeTest.Tests.csproj (hunk 1 above)
Comment (philipaconrad) [reply test/SmokeTest.Tests/SmokeTest.Tests.csproj:34]: My thoughts exactly! :slightly_smiling_face: Adding each individual file was quite a pain. At least now that's limited to just the test fixture code, and doesn't also involve the build system anymore.

**Comment by srenatus** [comment]
> (Used to spin up the EOPA test container.)

I think if we're passing `--no-license-fallback`, we should be OK to run the tests without a (valid) license. Am I mistaken?

**Comment by philipaconrad** [comment]
:information_source: Added some checks in the testcases for when we're in OPA fallback mode versus EOPA mode (Since that's definitely not intended in CI here). The error messages have the same intent, but they have very different message text when that happens.

OPA Fallback mode:
```
testdata/condfail.rego:5: eval_conflict_error: object keys must be unique
```
versus EOPA mode:
```
object insert conflict
```

**Pull request merged**